Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1

```diff
@@ -285,19 +288,44 @@ function Configure-ConditionalAccessPolicyForPIM
     if ([String]::IsNullOrWhiteSpace($FilePath))
     {
         $policyAssignments = Get-AzRoleManagementPolicyAssignment -Scope "subscriptions/$($SubscriptionId)"
-
+        
+        # Fetch all the role definitions (built-in and custom).
+        $roleDefinitions = Get-AzRoleDefinition | Select-Object -ExpandProperty Name
+        
         if (-not $RoleName)
-        {
-            $adminRoles = @("Owner", "Contributor", "User Access Administrator")
+        {   
+            # These are the roles which are excluded by the control.
+            $excludedRoles = @(
+                "Azure Front Door Domain Contributor",
+                "Azure Front Door Domain Reader",
+                "Azure Front Door Profile Reader",
+                "Azure Front Door Secret Contributor",
+                "Azure Front Door Secret Reader",
+                "Defender for Storage Data Scanner",
+                "AzureML Compute Operator",
+                "Cognitive Services Usages Reader",
+                "Key Vault Crypto Service Release User"
+            )
+
+            # Remove excluded controls from all role definition.
+            $criticalRoles = $roleDefinitions | Where-Object { $_ -notin $excludedRoles }
         }
         else
         {
-            $adminRoles = @($RoleName)
+            # Validate if RoleName is a valid for the particular subscription (role name can be subscription specific for custom roles).
+            if (-not $roleDefinitions.Contains($RoleName))
```

Comment:
should we check for builtin role as well.

Reply:
We are checking for all roles, built-in and custom, what the comment tries to say is, if roles can be subscription specific i.e. a role can be present in one subscription and might not be present in another subscription.